### ISSUE TITLE
a11y: Combobox haspopup value should be `listbox` & should have aria-controls attribute when expanded

### ISSUE THREAD

**Issue by Akshat55** [IBM/carbon-components-angular#2276]
## Detailed description

> Describe in detail the issue you're having.
Current aria-haspopup value is set to `true`. While this is correct, it should be changed to `listbox` which is more specific. If we take a look at the template, you can see that aria-haspopup attribute has been passed twice. Initially as `listbox` then `true`, hence overwriting `listbox` to `true`.

Once the `aria-haspopup` value has been updated to `listbox` & the accessibility checker is run, you can see that `aria-controls` is missing.

> Is this a feature request (new component, new icon), a bug, or a general issue?

Bug

> Is this issue related to a specific component?

Combobox & Dropdownlist

> What did you expect to happen? What happened instead? What would you like to see changed?

Should have correct role & necessary attributes to go with them

> What browser are you working in?

Firefox/Chrome

> What version of the Carbon Design System are you using?

4.61.0

**Comment by Akshat55** [comment]
Making this PR into next since integration for angular 8 is failing for some reason... Most likely has to do with styles package being updated 🤔
